File: app-android/app/src/main/java/de/tutao/tutanota/IdTuple.java
package de.tutao.tutanota;

public final class IdTuple {

    private final String listId;
    private final String elementId;

    public IdTuple(String listId, String elementId) {

        this.listId = listId;
        this.elementId = elementId;
    }

    public String getElementId() {
        return elementId;
    }

    public String getListId() {
        return listId;
    }
}

File: app-ios/tutanota/Sources/Alarms/IdTuple.swift
import Foundation

/// Id which includes list and element Id.
/// In JSON represented as an array tuple but structure is easier to use so we manually implement coding.
struct IdTuple {
  let listId: String
  let elementId: String
}

extension IdTuple : Encodable {
  func encode(to encoder: Encoder) throws {
    var container = encoder.unkeyedContainer()
    try container.encode(self.listId)
    try container.encode(self.elementId)
  }
}

extension IdTuple : Decodable {
  init(from decoder: Decoder) throws {
    var values = try decoder.unkeyedContainer()
    let listId = try values.decode(String.self)
    let elementId = try values.decode(String.self)
    
    self.init(listId: listId, elementId: elementId)
  }
}
